onnxruntime/core/graph/graph.cc

```diff
@@ -1494,6 +1494,15 @@ Status Graph::BuildConnections(std::unordered_set<std::string>& outer_scope_node
                 ORT_IGNORE_RETURN_VALUE(outer_scope_node_args_consumed.insert(input_arg_name));
               }
             }
+          } else {
+            if (resolve_context_.inputs_and_initializers.find(input_arg_name) ==
+                    resolve_context_.inputs_and_initializers.cend() &&
+                // if we're manually creating a Graph for use as a subgraph the outer scope names are manually set here
+                outer_scope_node_arg_names_.find(input_arg_name) ==
+                    outer_scope_node_arg_names_.cend()) {
+              return ORT_MAKE_STATUS(ONNXRUNTIME, FAIL, "Invalid model. Node input '", input_arg_name,
```

Comment:
Since the input to ORT is incorrect
```suggestion
              return ORT_MAKE_STATUS(ONNXRUNTIME, INVALID_ARGUMENT, "Invalid model. Node input '", input_arg_name,
```